pkg/apis/options/sessions.go

```diff
@@ -1,10 +1,14 @@
 package options
 
+import "time"
+
 // SessionOptions contains configuration options for the SessionStore providers.
 type SessionOptions struct {
-	Type   string             `flag:"session-store-type" cfg:"session_store_type"`
-	Cookie CookieStoreOptions `cfg:",squash"`
-	Redis  RedisStoreOptions  `cfg:",squash"`
+	Type       string             `flag:"session-store-type" cfg:"session_store_type"`
+	Retry      int                `flag:"session-retry-attempts" cfg:"session_retry_attempts"`
```

Comment:
What's the exact race scenario we are trying to protect against - I want to make sure we get the right configurable knobs without being overkill with too many flags:

Parallel requests A, B:
(A) has completed the refresh_token token_endpoint redeem flow (blocking request B from doing it)
(B) Comes in after (A) has done the refresh_token, but before it persists it to the session store?

If that's the scenario, what is the average time between refresh_token call & persisting to redis session store by (A). Are there sensible automatic retry amounts and an exponential backoff we can just use by default?

Reply:
I had vague notions of distributed locks... but I think with how we use Redis & other persistent session stores int he future, that doesn't fit with our existing architecture easily.

Reply:
Good point, I will try to figure that out and set the average value as waiting time. With retries you still have enough control.

As long term solution, I would also prefer a lock, but I think based on the current architecture this is the most simple and proper working solution.

Reply:
I think the time to talk to persist the fresh session after a response from the IdP is in the milliseconds order of magnitude.

If (B) is mostly parallel to (A), but it comes in second and fails with the IdP, it will still come back in the same manner to the proxy as (A) due to network RTT. So I think (A) response received -> Redis is the time we have to be aware of.

100ms-250ms seems sensible to me with that back of napkin math 😅